cl/users/tasks.py

```diff
@@ -22,57 +24,81 @@ def abort_or_retry(task, exc):
 
-    Perform update if email address is already registered.
+    Args:
+        user_id (int): The ID of the user to check and potentially create an
+        account for.
 
-    :param self: The celery task
-    :param email: The user's email address
-    :param action: Action to perfom on moosend
-    :return: None
+    Raises:
+        AssertionError: If more than one matching account is found in Neon CRM.
     """
-    allowed_actions = ["subscribe", "unsubscribe"]
-    assert action in allowed_actions, f"'{action}' is not an allowed action."
-    params = {"apikey": settings.MOOSEND_API_KEY}  # type: ignore
-
-    if action == "subscribe":
-        path = f"/v3/subscribers/{settings.MOOSEND_DEFAULT_LIST_ID}/subscribe.json"  # type: ignore
-    else:
-        path = f"/v3/subscribers/{settings.MOOSEND_DEFAULT_LIST_ID}/unsubscribe.json"  # type: ignore
-
-    try:
-        r = requests.post(
-            url=urljoin(settings.MOOSEND_API_URL, path),  # type: ignore
-            params=params,
-            json={
-                "Email": email,
-            },
-            timeout=30,
+    user = User.objects.select_related("profile").get(pk=user_id)
+    neon_client = NeonClient()
+    neon_accounts = neon_client.search_account_by_email(user.email)
+
+    if len(neon_accounts) > 1:
+        # Neon CRM automatically merges accounts with identical names and
+        # email addresses. This process, called Account Match feature, ensures
+        # consistent data across records. However, If the accounts are very
+        # close (email and phone match but not name) the accounts will be
+        # entered into the Partial Match Queue. This queue allows users to
+        # review these potential matches and decide whether to merge them
+        # manually.
+        raise AssertionError(
+            "There's more than one account using the same email address. "
+            "We should check the Partial Match Queue."
         )
-    except requests.RequestException as exc:
-        abort_or_retry(self, exc)
-        return
 
-    j = r.json()
-    code = j.get("Code")
+    profile = user.profile  # type: ignore
+    if len(neon_accounts) == 1:
+        # We found an existing account that matches the email address. we'll
+        # use that one instead of creating a new one to avoid potential future
+        # merges.
+        profile.neon_account_id = neon_accounts[0]["Account ID"]
+        profile.save()
+        return None
 
-    if code == 0:
-        logger.info(
-            "Successfully completed '%s' action on '%s' in moosend.",
-            action,
-            email,
-        )
-    else:
-        error = j.get("Error", "Unknown error")
-        logger.warning(
-            "Did not complete '%s' action on '%s' in moosend: '%s'",
-            action,
-            email,
-            error,
-        )
+    new_account_id = neon_client.create_account(user)
+    profile.neon_account_id = new_account_id
+    profile.save()
```

Comment:
To make you code more self-documenting, I'd put this under an if statement: 

```
if num_neon_accounts_found == 0:
    # No account found, create one
```

Reply:
Thanks for the suggestion. I will update the code accordingly.